lib/magma/server/update.rb

```diff
@@ -55,4 +57,36 @@ def load_revisions
     log(m.complaints)
     @errors.concat(m.complaints)
   end
+
+  def update_any_file_links
+    # Here, if there are any File attributes in the revisions and
+    #   the storage is configured to use Metis instead of AWS,
+    #   we'll send a request to the Metis "copy" route, using the
+    #   Etna::Client.
+    if Magma.instance.config(:storage).fetch(:provider).downcase == 'metis'
+
+      # Etna::Client.new throws an exception if the host config includes protocol,
+      # i.e. https://metis-dev.etna-development.org will throw an exception
+      raise Etna::BadRequest, 'Storage host should not include protocol' if
+        Magma.instance.config(:storage).fetch(:host).start_with? 'http'
+
+      # Do we get the token from config?
+      client = Etna::Client.new(
+        "https://#{Magma.instance.config(:storage).fetch(:host)}",
+        Magma.instance.config(:token) ? Magma.instance.config(:token) : 'token')
+
+      @revisions.each do |model, model_revisions|
+        model_revisions.each do |revision|
+          if revision.is_a?(File)
+            # copy_url = Magma.instance.storage.copy_url(
+            #   @project_name,
+            #   revision. # Where do I just pull out the path from?
+            # )
+
+            # Use the Etna::Client to make a POST call to Metis's copy_url?
```

Comment:
I'm not sure how well Etna::Client will play with a full URL, since its helpers expect the route name and try to build the request path for you. So we need to do something like:
  1) Build the route path for the copy request (probably just with a string template)
  2) Use Etna::Hmac to compute the hmac params for that path
  3) Use Etna::Client to request that path + hmac params (probably circumventing the copy route's helper method and using Etna::Client#post or something)

Reply:
Okay, sounds good. So I don't need a separate `copy_url()` method that computes the full URL path + HMAC values.